smartfs.lua

```diff
@@ -326,31 +341,68 @@ function smartfs._makeState_(form, newplayer, params, is_inv, nodepos)
 				meta:set_string("smartfs_name", self.def.name)
 			end
 		end,
-		-- Receive fields and actions from formspec
-		_sfs_on_receive_fields_ = function(self, player, fields)
-			assert(self)
-			assert(player)
-
-			for key,val in pairs(fields) do
-				if self._ele[key] then
-					self._ele[key].data.value = val
+		-- process /apply received field value
+		_sfs_process_value_ = function(self, field, value)
```

Comment:
the difference between value and action seems weird to me. You should combine them into one. Might even be worth calling on_receive_fields one for each container instead of recursing for each field